components/brave_wallet/browser/zcash/zcash_shield_sync_service.cc

```diff
@@ -443,9 +441,9 @@ ZCashRpc& ZCashShieldSyncService::zcash_rpc() {
   return zcash_wallet_service_->zcash_rpc();
 }
 
-base::SequenceBound<ZCashOrchardStorage>&
-ZCashShieldSyncService::orchard_storage() {
-  return zcash_wallet_service_->orchard_storage();
+base::SequenceBound<ZCashOrchardSyncState>&
+ZCashShieldSyncService::sync_state() {
+  return zcash_wallet_service_->sync_state();
```

Comment:
You should not be passing the wallet service into the constructor just to get access `ZCashOrchardSyncState` and `ZCashRpc`. This violates chromium design principles for modularity https://chromium.googlesource.com/chromium/src/+/main/docs/chrome_browser_design_principles.md
pass `ZCashOrchardSyncState` and `ZCashRpc` in directly

Reply:
We have an [issue](https://github.com/brave/brave-browser/issues/42022) for this

Reply:
but why are you adding to the problem? This is a trivial fix here and enables easier unit testing

Reply:
Switched to direct references